pylint/checkers/imports.py

```diff
@@ -512,39 +514,44 @@ def _check_imports_order(self, node):
 
         Imports must follow this order: standard, 3rd party, local
         """
-        extern_imports = []
-        local_imports = []
-        std_imports = []
+        imports = {}
+        isort2keys = {
+            'FUTURE': 'standard',
+            'STDLIB': 'standard',
+            'FIRSTPARTY': 'external',
+            'THIRDPARTY': 'external',
+            'LOCALFOLDER': 'local',
+        }
         isort_obj = isort.SortImports(
             file_contents='', known_third_party=self.config.known_third_party,
             known_standard_library=self.config.known_standard_library,
         )
         for node, modname in self._imports_stack:
-            if modname.startswith('.'):
-                package = '.' + modname.split('.')[1]
-            else:
-                package = modname.split('.')[0]
-
-            import_category = isort_obj.place_module(package)
-            if import_category in ('FUTURE', 'STDLIB'):
-                std_imports.append((node, package))
-                wrong_import = extern_imports or local_imports
+            package = '.' + modname.split('.')[1] if modname.startswith('.') \
+                else modname.split('.')[0]
+            isort_category = isort_obj.place_module(package)
+            category = isort2keys.get(isort_category, '')
+            # to share the same pointer of memory in list.append sentences
+            node_pack = (node, package)
+            imports.setdefault(category + '_all', []).append(node_pack)
```

Comment:
I think this code is overly complicated and I can't figure out what you're intending to do with it. First of all, you're appending _nested to the category, but I can't see where you're removing it, users shouldn't care about being _nested or _all. Second, I feel that the use of setdefault is superfluous, why don't you simply determine the imports using some variables instead of putting the nodes into a dictionary?